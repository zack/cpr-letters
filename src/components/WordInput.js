import React from 'react';
import { FormControl, FormGroup, InputGroup } from 'react-bootstrap';

const WordInput = ({words, value, letter, changeHandler}) => {
  let validationState;

  if (words.includes(value)) {
    validationState = 'success';
  } else if (value) {
    validationState = 'warning';
  } else {
    validationState = 'error';
  }

  return (
    <FormGroup validationState={validationState}>
      <InputGroup>
        <InputGroup.Addon>
          <strong>{letter.toUpperCase()}</strong>
        </InputGroup.Addon>
        <FormControl type="text" onChange={changeHandler}/>
      </InputGroup>
    </FormGroup>
  );
}

export default WordInput;
