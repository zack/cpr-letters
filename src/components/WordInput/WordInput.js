import React from 'react';
import { FormControl, FormGroup, InputGroup } from 'react-bootstrap';

var getValidationState = (word, value) => {
  if (new RegExp(word).test(value)) {
    return('success');
  } else if (value) {
    return('warning');
  } else {
    return(null);
  }
}

const WordInput = ({word, value, letter, changeHandler}) => {
  return (
    <FormGroup validationState={getValidationState(word, value)}>
      <InputGroup>
        <InputGroup.Addon>
          <strong>{letter.toUpperCase()}</strong>
        </InputGroup.Addon>
        <FormControl type="text" onChange={changeHandler}/>
      </InputGroup>
    </FormGroup>
  );
}

export default WordInput;
