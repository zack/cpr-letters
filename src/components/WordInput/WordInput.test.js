import React from 'react';
import WordInput from './WordInput.js';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('<WordInput />', () => {
  it('Reflects the input in its style', () => {
    ['', 't', 'test'].forEach((value) => {
      const component = renderer.create(
        <WordInput word='test' value={value} letter='t' />
      );
      expect(component).toMatchSnapshot();
    });
  });

  it('Calls its changeHandler onChange of the inner FormControl', () => {
    const onChange = jest.fn();
    const component = mount(
      <WordInput word='test' value='' letter='t' changeHandler={onChange} />
    );

    const input = component.find('input').first();
    input.simulate('change');
    expect(onChange).toBeCalled();
  });
});
