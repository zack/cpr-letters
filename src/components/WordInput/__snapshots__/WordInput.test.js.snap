// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WordInput /> Reflects the input in its style 1`] = `
<div
  className="form-group"
>
  <span
    className="input-group"
  >
    <span
      className="input-group-addon"
    >
      <strong>
        T
      </strong>
    </span>
    <input
      className="form-control"
      id={undefined}
      onChange={undefined}
      type="text"
    />
  </span>
</div>
`;

exports[`<WordInput /> Reflects the input in its style 2`] = `
<div
  className="form-group has-warning"
>
  <span
    className="input-group"
  >
    <span
      className="input-group-addon"
    >
      <strong>
        T
      </strong>
    </span>
    <input
      className="form-control"
      id={undefined}
      onChange={undefined}
      type="text"
    />
  </span>
</div>
`;

exports[`<WordInput /> Reflects the input in its style 3`] = `
<div
  className="form-group has-success"
>
  <span
    className="input-group"
  >
    <span
      className="input-group-addon"
    >
      <strong>
        T
      </strong>
    </span>
    <input
      className="form-control"
      id={undefined}
      onChange={undefined}
      type="text"
    />
  </span>
</div>
`;
